pkg/cloudprovider/vm_test.go

```diff
@@ -0,0 +1,49 @@
+package cloudprovider
+
+import (
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-12-01/compute"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSetUserIdentitiesVM(t *testing.T) {
+	testIdentityInfo := &vmIdentityInfo{
+		info: &compute.VirtualMachineIdentity{},
+	}
+
+	// adding id1
+	update := testIdentityInfo.SetUserIdentities(map[string]bool{"id1": true})
+	assert.True(t, update)
+	// adding id2
+	update = testIdentityInfo.SetUserIdentities(map[string]bool{"id2": true})
+	assert.True(t, update)
+	// add id3 and delete id1
+	update = testIdentityInfo.SetUserIdentities(map[string]bool{"id3": true, "id4": true, "id1": false})
+	assert.True(t, update)
+}
+
+func TestRemoveUserIdentityVM(t *testing.T) {
+	testIdentityInfo := &vmIdentityInfo{
+		info: &compute.VirtualMachineIdentity{
+			UserAssignedIdentities: map[string]*compute.VirtualMachineIdentityUserAssignedIdentitiesValue{
+				"ID1": nil,
```

Comment:
Lets change the value to `&compute.VirtualMachineIdentityUserAssignedIdentitiesValue`